include/alpaka/pltf/PltfGenericSycl.hpp

```diff
@@ -0,0 +1,440 @@
+                        }
+                    }
+                };
+
+#    pragma clang diagnostic push
+#    pragma clang diagnostic ignored "-Wexit-time-destructors"
+                static auto pf = platform{typename TPltf::selector{}};
+                static auto devices = pf.get_devices();
+                static auto ctx = context{devices, exception_handler};
+#    pragma clang diagnostic pop
+
+                if(devIdx >= devices.size())
+                {
+                    auto ss_err = std::stringstream{};
+                    ss_err << "Unable to return device handle for device " << devIdx << ". There are only "
+                           << devices.size() << " SYCL devices!";
+                    throw std::runtime_error(ss_err.str());
+                }
+
+                auto sycl_dev = devices.at(devIdx);
+
+                // Log this device.
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                printDeviceProperties(sycl_dev);
+#    elif ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+                std::cout << __func__ << sycl_dev.get_info<info::device::name>() << '\n';
+#    endif
+                return typename DevType<TPltf>::type{sycl_dev, ctx};
+            }
+
+        private:
+#    if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+            //! Prints all the device properties to std::cout.
+            static auto printDeviceProperties(const sycl::device& device) -> void
+            {
+                ALPAKA_DEBUG_FULL_LOG_SCOPE;
+
+                constexpr auto KiB = std::size_t{1024};
+                constexpr auto MiB = KiB * KiB;
+
+                std::cout << "Device type: ";
+                switch(device.get_info<sycl::info::device::device_type>())
+                {
+                case sycl::info::device_type::cpu:
+                    std::cout << "CPU";
+                    break;
+
+                case sycl::info::device_type::gpu:
+                    std::cout << "GPU";
+                    break;
+
+                case sycl::info::device_type::accelerator:
+                    std::cout << "Accelerator";
+                    break;
+
+                case sycl::info::device_type::custom:
+                    std::cout << "Custom";
+                    break;
+
+                case sycl::info::device_type::automatic:
+                    std::cout << "Automatic";
+                    break;
+
+                case sycl::info::device_type::host:
+                    std::cout << "Host";
+                    break;
+
+                // The SYCL spec forbids the return of device_type::all
+                // Including this here to prevent warnings because of
+                // missing cases
+                case sycl::info::device_type::all:
+                    std::cout << "All";
+                    break;
+                }
+                std::cout << std::endl;
+
+                std::cout << "Name: " << device.get_info<sycl::info::device::name>() << std::endl;
```

Comment:
```suggestion
                std::cout << "Name: " << device.get_info<sycl::info::device::name>() << '\n';
```
And then only do the flush at the end of the function.